crates/ide/src/expand_macro.rs

```diff
@@ -33,6 +34,18 @@ pub(crate) fn expand_macro(db: &RootDatabase, position: FilePosition) -> Option<
     let mut expanded = None;
     let mut name = None;
     for node in tok.ancestors() {
+        if let Some(attr) = ast::Attr::cast(node.clone()) {
+            if let Some((path, tt)) = attr.as_simple_call() {
+                if path == "derive" {
```

Comment:
Technically you can refer to the derive macro by its full prelude path but I'd be surprised if anyone does that(also some other places just assume it like this I believe)